oracles/src/repositories/oracle_repository.py

```diff
@@ -220,6 +255,30 @@ async def send_function_call_response(
         function_call.is_processed = bool(tx_receipt.get("status"))
         return bool(tx_receipt.get("status"))
 
+    async def mark_function_call_as_done(self, function_call: FunctionCall):
+        nonce = await self.web3_client.eth.get_transaction_count(self.account.address)
+        tx_data = {
+            "from": self.account.address,
+            "nonce": nonce,
+            # TODO: pick gas amount in a better way
+            # "gas": 1000000,
+            "maxFeePerGas": self.web3_client.to_wei("2", "gwei"),
+            "maxPriorityFeePerGas": self.web3_client.to_wei("1", "gwei"),
+        }
+        if chain_id := settings.CHAIN_ID:
+            tx_data["chainId"] = int(chain_id)
+
+        tx = await self.oracle_contract.functions.markFunctionAsProcessed(
+            function_call.id,
+        ).build_transaction(tx_data)
+        signed_tx = self.web3_client.eth.account.sign_transaction(
```

Comment:
We should probably make a function to do the signing and sending tx, a lot of copy paste happening

Reply:
Cool! Will do that